internal/service/ipam/ipam.go

```diff
@@ -0,0 +1,307 @@
+/*
+Copyright 2024 IONOS Cloud.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ipam offers services for IPAM management.
+package ipam
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ipamv1 "sigs.k8s.io/cluster-api/exp/ipam/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	infrav1 "github.com/ionos-cloud/cluster-api-provider-ionoscloud/api/v1alpha1"
+	"github.com/ionos-cloud/cluster-api-provider-ionoscloud/scope"
+)
+
+const (
+	// PrimaryNICFormat is the format used for IPAddressClaims for the primary nic.
+	PrimaryNICFormat = "nic-%s"
+
+	// AdditionalNICFormat is the format used for IPAddressClaims for additional nics.
+	AdditionalNICFormat = "nic-%s-%d"
+
+	// IPV4Format is the IP v4 format.
+	IPV4Format = "ipv4"
+
+	// IPV6Format is the IP v6 format.
+	IPV6Format = "ipv6"
```

Comment:
Is there a need to export these consts to other packages?

Reply:
no, i'll make them private